pipeline_products_session/jwst-data-products-part1.ipynb

```diff
@@ -0,0 +1,961 @@
+    "\n",
+    "# or \n",
+    "\n",
+    "print('\\nNumber of data dimensions: ', science_headers['NAXIS'])\n",
+    "print('Number of integrations: ', science_headers['NAXIS4'])\n",
+    "print('Number of groups: ', science_headers['NAXIS3'])\n",
+    "print('Number of rows: ', science_headers['NAXIS2'])\n",
+    "print('Number of columns: ', science_headers['NAXIS1'])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Additional metadata is stored in the ASDF extension. This extension can be read using The Advanced Scientific Data Format (ASDF), which is a next-generation format for scientific data. ASDF is a tool for reading and writing ASDF files. More information about the ASDF file standard is in the [ASDF software readthedocs](https://asdf.readthedocs.io/en/stable/). The format has the following features:\n",
+    "\n",
+    "* A hierarchical, human-readable metadata format (implemented using YAML)\n",
+    "* Numerical arrays are stored as binary data blocks which can be memory mapped. Data blocks can optionally be compressed.\n",
+    "* The structure of the data can be automatically validated using schemas (implemented using JSON Schema)\n",
+    "* Native Python data types (numerical types, strings, dicts, lists) are serialized automatically\n",
+    "* ASDF can be extended to serialize custom data types\n",
+    "\n",
+    "Right now, you don't need to worry about ASDF too much. We'll talk about it more when we discuss configuration files and accessing the WCS information in the following modules. Below, we provide a simple example of how to access the ASDF extension:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "with fits.open(uncal_obs) as hdu:\n",
+    "    asdf_metadata = hdu['ASDF'].header\n",
+    "    asdf_data = hdu['ASDF'].data    "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "asdf_metadata"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "asdf_data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### C.<font color='white'>-</font>Visualizing data<a class=\"anchor\" id=\"astro-viz\"></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In the previous section, we mentioned [up-the-ramp sampling](https://jwst-docs.stsci.edu/understanding-exposure-times) for IR detectors. During an integration, the detectors accumulate charge while being read out multiple times non-destructively according to readout patterns. The readout process is non-destructive, leaving charge unaffected and in place (charge is not transferred between pixels as in CCDs). After each integration, the pixels are read out a final time and then reset, releasing their charge. \n",
```

Comment:
while being read out multiple times non-destructively according to readout patterns -> while being read out multiple times non-destructively following predefined readout patterns
and then link to the JDox readout pattern page? 
https://jwst-docs.stsci.edu/near-infrared-camera/nircam-instrumentation/nircam-detector-overview/nircam-detector-readout-patterns

Reply:
Changed to "During an integration, the detectors accumulate charge while being read out multiple times following predefined readout patterns for the different instruments. The readout process is non-destructive, leaving charge unaffected and in place (charge is not transferred between pixels as in CCDs)."